### ISSUE TITLE
GFLOPs and  GFLOPS

### ISSUE THREAD

**Issue by cainiao123s** [ultralytics/ultralytics#22324]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, during my research, I found that many people are not clear about the concepts of GFLOPs and GFLOPS. After I examined the source code and read the literature, I think the GFLOPs metric provided in YOLO represents the total number of floating-point operations required for the model to complete one forward pass at a specified input resolution. GFLOPS, on the other hand, represents the number of floating-point operations performed per second. In addition, is TOPs equal to 2 times GFLOPS?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cainiao123s, thanks for opening this discussion and for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also follow up with you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples and where many common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (as it appears here), please share any relevant details that could help clarify your context, such as:
-Exact Ultralytics version and environment (CPU/GPU, CUDA/CUDNN, PyTorch)
-Model used (e.g., YOLO11n/s/m/l/x) and input image size
-How you’re obtaining or interpreting the metrics (CLI output, Python API, logs)
-Any performance measurements or comparisons you’re making

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please ensure you’re on the latest `ultralytics` release with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify any discrepancies aren’t already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Dex947** [comment]
Hi @cainiao123s! Great question about the distinction between GFLOPs and GFLOPS.

You're absolutely correct in your understanding:

## **GFLOPs (Giga Floating Point Operations)**
- Represents the **total number** of floating-point operations required for one forward pass
- It's a measure of **computational complexity** of the model
- Unit: billions of operations (static metric)
- Example: "YOLOv8n requires 8.7 GFLOPs per inference"

## **GFLOPS (Giga Floating Point Operations Per Second)**
- Represents the **rate** of floating-point operations performed
- It's a measure of **computational throughput** or processing speed
- Unit: billions of operations per second (performance metric)
- Example: "This GPU can perform 10 TFLOPS (10,000 GFLOPS)"

## **Regarding TOPs vs GFLOPS**

Your question about TOPs (Tera Operations Per Second) is interesting:
- **TOPs** typically refers to **integer operations** (INT8/INT4), commonly used in quantized neural networks
- **GFLOPS** refers to **floating-point operations** (FP32/FP16)
- The relationship isn't exactly 2:1 - it depends on the data type and hardware architecture

**For quantized models:**
- INT8 operations can be 2-4x faster than FP32 on specialized hardware
- So 1 TOPS ≈ 1000 GFLOPS in throughput, but they measure different operation types

**Key Takeaway:** GFLOPs measures model complexity (how many operations needed), while GFLOPS/TOPS measure hardware performance (how fast operations are executed).

Hope this clarifies the concepts!
